walletunlocker/service.go

```diff
@@ -119,6 +125,14 @@ func New(chainDir string, params *chaincfg.Params, noFreelistSync bool,
 	}
 }
 
+// Stop waits for all requests to be handled, then closes all channels in
+// preparation for the wrapping gRPC server to be shut down.
+func (u *UnlockerService) Stop() {
+	u.requestWg.Wait()
+	close(u.InitMsgs)
```

Comment:
Why are these closed exactly?

Reply:
I'm not sure it is safe either, as there are no guarantee a send on the chanenl isn't attempted after they have been closed.

Reply:
Right, was an unnecessary with potentially unwanted side effects. Removed the close.